src/Units/KaijuWarsWeapons.java

```diff
@@ -0,0 +1,264 @@
+package Units;
+
+import Terrain.TerrainType;
+import Units.KaijuWarsUnits.KaijuWarsUnitModel;
+
+public class KaijuWarsWeapons
+{
+  // Multiply all movement (and ranges) by N
+  public static final int KAIJU_SCALE_FACTOR = 1;
+  // Percent damage that 1 ATK should do vs 1 kaijuCounter
+  protected final static int KAIJU_DAMAGE_FACTOR = 80;
+  protected final static int KAIJU_DAMAGE_BASE = 55;
+
+  private static class KaijuWarsWeapon extends WeaponModel
+  {
+    private static final long serialVersionUID = 1L;
+    private final static boolean infiniteAmmo = true;
+
+    public boolean isAirWeapon          = false;
+    public boolean negateCounterBonuses = false;
+    public final int vsLand, vsAir;
+    protected KaijuWarsWeapon(int vsLand, int vsAir, int minRange, int maxRange)
+    {
+      super(infiniteAmmo, minRange, maxRange);
+      this.vsLand = vsLand;
+      this.vsAir  = vsAir;
+      if( maxRange > 1 )
+        negateCounterBonuses = true;
+    }
+    protected KaijuWarsWeapon(int vsLand, int vsAir)
+    {
+      this(vsLand, vsAir, 1, 1);
+    }
+    public KaijuWarsWeapon(KaijuWarsWeapon other)
+    {
+      this(other.vsLand, other.vsAir, other.rangeMin, other.rangeMax);
+      canFireAfterMoving = other.canFireAfterMoving;
+    }
+    @Override
+    public WeaponModel clone()
+    {
+      return new KaijuWarsWeapon(this);
+    }
+
+    protected final static int SLOW_BONUS = 2;
+    @Override
+    public double getDamage(KaijuWarsUnitModel defender)
+    {
+      int counterPower = defender.kaijuCounter;
+      if( !negateCounterBonuses )
+      {
+        if( isAirWeapon && defender.slowsAir )
+          counterPower += SLOW_BONUS;
+        if( !isAirWeapon && defender.slowsLand )
+          counterPower += SLOW_BONUS;
+        if( defender.resistsKaiju )
+          counterPower += SLOW_BONUS;
+      }
+
+      int attack = 0;
+      if( defender.isAirUnit() )
+        attack = vsAir;
+      else
+        attack = vsLand;
```

Comment:
Aww, missed a perfect opportunity for a ternary operator.

Reply:
I considered it, FWIW.